build/elasticsearch/bin/es-docker

```diff
@@ -22,4 +22,6 @@ do
     fi
 done < <(env)
 
+echo "-Des.cgroups.hierarchy.override=/" >> config/jvm.options
```

Comment:
> Just so that I understand better this solution, basically `es.cgroups.hierarchy.override` can be defined as a jvm option, as documented [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/setting-system-settings.html#jvm-options) hence will be honored when presented through `ES_JAVA_OPTS`?

Yes.

> I am asking this as [my understanding](https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html#_setting_default_settings) is that most parameters defined in `jvm.options` can also be passed as `-E` style cli args. If that is true, the user can run the image using `docker run ... -e es.cgroups.hierarchy.override=/` and so it could be the value [here](https://github.com/elastic/elasticsearch-docker/blob/master/build/elasticsearch/bin/es-docker#L11) instead of the existing `es_opts=''`?

This part is not right, `ES_JAVA_OPTS` and `jvm.options` control JVM flags, so options in `ES_JAVA_OPTS` can also be set in `jvm.options` and vice-versa. The `-E` CLI flags to Elasticsearch are Elasticsearch settings and can also be set in `elasticsearch.yml`.

> (Note that I tried the last point with 6.0.0-alpha1-SNAPSHOT and received:
>```
>uncaught exception in thread [main]
>org.elasticsearch.bootstrap.StartupException: java.lang.IllegalArgumentException: unknown >setting [es.cgroups.hierarchy.override] please check that any required plugins are installed, or >check the breaking changes documentation for removed settings
>```
>)

This is expected since `es.cgroups.hierarchy.override` is not an Elasticsearch setting, we are reading it as JVM system property which is set via the JVM flag `-Des.cgroups.hierarchy.override=/`; this can be done via `jvm.options` or `ES_JAVA_OPTS`.